Framework/DataObjects/src/PeaksWorkspace.cpp

```diff
@@ -149,6 +150,28 @@ void PeaksWorkspace::removePeak(const int peakNum) {
   peaks.erase(peaks.begin() + peakNum);
 }
 
+/** Removes multiple peaks
+* @param badPeaks peaks to be removed
+*/
+void PeaksWorkspace::removePeaks(std::vector<int> &badPeaks) {
+  if (badPeaks.size() == 0)
+    return;
+  auto first = peaks.begin();
+  auto end = peaks.end();
+  auto result = first;
+  std::sort(badPeaks.begin(), badPeaks.end());
+  for (int i = 0; first < end; ++first, ++i) {
+    // if index of peak is not in badPeaks
+    if (!std::binary_search(badPeaks.begin(), badPeaks.end(), i)) {
+      // include in result
+      *result = std::move(*first);
```

Comment:
Could we rename `first` - I keep looking at it thinking we are prepending the first element to the result iterator. Maybe something like `index`?